filecheck/checker.ts

```diff
@@ -290,48 +277,100 @@ async function checkCompression(filePath: string, options: CheckerOptions) {
   }
 }
 
-function canCheckFile(filePath: string) {
-  const filePathParts = filePath.split(path.sep);
-
+function isRelevantFile(filePath: string) {
   return (
-    filePathParts.includes("files") &&
-    !filePathParts.includes("node_modules") &&
-    !/\.(DS_Store|html|json|md|txt|yml)$/i.test(filePath)
+    filePath.includes(`files${path.sep}`) &&
+    !/\.(DS_Store|ini)$/i.test(filePath)
   );
 }
 
+function isBinaryFile(filePath: string) {
+  return !/\.(html|json|md|txt|yml)$/i.test(filePath);
+}
+
 async function resolveDirectory(file: string): Promise<string[]> {
   const stats = await fs.lstat(file);
   if (stats.isDirectory()) {
     const api = new fdir()
       .withErrors()
       .withFullPaths()
-      .filter((filePath) => canCheckFile(filePath))
+      .filter((filePath) => isRelevantFile(filePath))
       .crawl(file);
     return api.withPromise() as Promise<PathsOutput>;
-  } else if (stats.isFile() && canCheckFile(file)) {
+  } else if (stats.isFile() && isRelevantFile(file)) {
     return [file];
   } else {
     return [];
   }
 }
 
+function validatePath(filePath: string): string[] {
+  const errors = [];
+
+  const shortPath = path.join(
+    path.basename(path.dirname(filePath)),
+    path.basename(filePath)
+  );
+
+  // Must have supported extension
+  const supportedFilesRegExp = /\.(md|json|txt|yml|jpg|jpeg|png|gif|svg)$/i;
+  if (!supportedFilesRegExp.test(filePath)) {
+    errors.push(
+      `Error: Invalid file: ${shortPath}. The file extension is not supported.`
+    );
+  }
+
+  // All characters must be lower case.
+  if (shortPath !== shortPath.toLowerCase() && !filePath.endsWith(".json")) {
+    errors.push(
+      `Error: Invalid path: ${shortPath}. All characters must be lowercase.`
+    );
+  }
+
+  // Whitespaces are not allowed.
+  if (filePath.includes(" ")) {
+    errors.push(
+      `Error: Invalid path: ${shortPath}. File path must not include whitespaces.`
+    );
+  }
+  if (filePath.includes("\u200b")) {
+    errors.push(
+      `Error: Invalid path: ${shortPath}. File path must not include zero width whitespaces.`
+    );
+  }
+
+  // File path should't contain banned characters: `(`, `)`
+  const bannedCharsRegExp = /[()]/;
+  if (bannedCharsRegExp.test(filePath)) {
+    errors.push(
+      `Error: Invalid path: ${shortPath}. File path must not include characters: '(', ')'`
+    );
+  }
+
+  return errors;
```

Comment:
We have more banned chars (see https://www.npmjs.com/package/sanitize-filename). If we start validating names we should do it the same way as for content paths.

Much more important we should not disallow special chars in the path to the content root (this would fail if I have content checked out to `/tmp/()/`).

Also we log the shortPath but check the filePath

Reply:
> We have more banned chars (see [npmjs.com/package/sanitize-filename](https://www.npmjs.com/package/sanitize-filename)). If we start validating names we should do it the same way as for content paths.

Used sanitised-filename library at the end of all checks.

> Much more important we should not disallow special chars in the path to the content root (this would fail if I have content checked out to `/tmp/()/`).
>
> Also we log the shortPath but check the filePath

It was a bug in the code. Now the code applies checks to `shortPath` and logs `filePath`.